### PR TITLE
[FIRRTL] Add path op parsing

### PR DESCRIPTION
**Pull request by youngar**
This adds parsing for path operations in to unresolved paths.

### CODE REVIEW

File: lib/Dialect/FIRRTL/Import/FIRParser.cpp [hunk 1]
```diff
@@ -1821,6 +1822,14 @@ ParseResult FIRStmtParser::parseExpImpl(Value &result, const Twine &message,
       return failure();
     break;
   }
+  case FIRToken::lp_path:
+    if (isLeadingStmt)
+      return emitError("unexpected path() as start of statement");
+    if (requireFeature({3, 2, 0}, "paths"))
+      return failure();
+    if (parsePathExp(result))
+      return failure();
```
Comment (rwy7) [inline lib/Dialect/FIRRTL/Import/FIRParser.cpp:1831]:
```suggestion
    if (requireFeature({3, 2, 0}, "paths") || parsePathExp(result))
      return failure();
```

**Review by rwy7 — APPROVED** [review]

File: lib/Dialect/FIRRTL/Import/FIRParser.cpp (hunk 1 above)
Comment (youngar) [reply lib/Dialect/FIRRTL/Import/FIRParser.cpp]: ty

File: lib/Dialect/FIRRTL/Import/FIRParser.cpp [hunk 2]
```diff
@@ -2967,6 +2974,23 @@ ParseResult FIRStmtParser::parseStaticRefExp(Value &result,
...
+ParseResult FIRStmtParser::parsePathExp(Value &result) {
+  auto startTok = consumeToken(FIRToken::lp_path);
+  locationProcessor.setLoc(startTok.getLoc());
+  StringRef target;
+  if (parseGetSpelling(target) ||
+      parseToken(FIRToken::string,
+                 "expected target string in path expression") ||
+      parseToken(FIRToken::r_paren, "expected ')' in path expression"))
+    return failure();
+  locationProcessor.setLoc(startTok.getLoc());
```
Comment (dtzSiFive) [inline lib/Dialect/FIRRTL/Import/FIRParser.cpp:2988]: Probably don't need to call `setLoc()` both here and above, don't think any of the methods between the two create operations or interact with that tracking?

**Review by fabianschuiki — APPROVED** [review]
LGTM!

**Review by dtzSiFive — COMMENTED** [review]
Parsing looks fine, added a comment there.

Please add some test to `parse-basic.mlir` and an error test of some kind?

**Comment by youngar** [comment]
Forgot to `git add` the parse-basic.fir test - also added a bunch of error tests.

**Pull request merged**